### PR TITLE
Keep n biggest parts of each feature and not layer

### PR DESCRIPTION
**Pull request by DelazJ**
Fix https://issues.qgis.org/issues/20840

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -1355,15 +1355,13 @@ See also
 
 Keep n biggest parts
 --------------------
-Cuts the n biggest parts of the input layer.
-
-This algorithm is particularly useful if a single layer is very complicated and
-made of many different parts.
+Takes a polygon layer and creates a new polygon layer in which multipart geometries
+have been removed, leaving only up to n largest (in terms of area) parts for each feature.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
have been removed, leaving only the *n* largest (in terms of area) parts for each feature.
```

Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Will the the resulting geometries also be multipart geometries (with only the n biggest parts kept), or will the resulting layer have up to n (non-multi) geometry features for each feature of the input multipart geometry layer?

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -1355,15 +1355,13 @@ See also
...
 
 .. figure:: img/n_biggest.png
    :align: center
 
-   Clockwise from left-up: source layer, one, tow and three biggest parts to keep
+   Clockwise from left-up: original multipart feature, one, two and three biggest parts to keep
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
   Clockwise from left-up: original multipart feature, one, two and three biggest parts kept
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -1373,15 +1371,15 @@ Parameters
 
 ``To keep`` [number]
   Choose how many biggest parts have to be kept. If 1 is selected, only the
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
  Choose how many parts to keep. If 1 is selected, only the
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 4]
```diff
@@ -1373,15 +1371,15 @@ Parameters
...
-  biggest part of the whole layer will be saved.
+  biggest part of the feature will be saved.
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
  biggest part of the feature will be kept.
```

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 1 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: > Will the the resulting geometries also be multipart geometries (with only the n biggest parts kept)

In my understanding, it's that. If the original layer has 10 features, you get 10 features. If you ask for 3 parts, then for each feature, you get 1, 2 or 3 parts depending on its original number of parts (reason why the *up to n*).
@nyalldawson ?

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 5]
```diff
@@ -1355,15 +1355,13 @@ See also
 
 Keep n biggest parts
 --------------------
-Cuts the n biggest parts of the input layer.
-
-This algorithm is particularly useful if a single layer is very complicated and
-made of many different parts.
+Takes a polygon layer and creates a new polygon layer in which multipart geometries
```
Comment (havatv) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]:
```suggestion
Takes a multipolygon layer and creates a new multipolygon layer in which the smallest polygons
```

Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I don't want to be nitpicking (I promised I stopped) but no, "**smallest** parts of each multipolygon" are not removed. The first and main idea is to "keep the largest parts". Then, **if** there are smaller parts, they are removed. The logic is to keep the bigger ones. so this is what should be expressed first imo. Not all the features have >n parts...

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 1 above)
Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: OK with the second sentence

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: I agree. Another go...

**Comment by havatv** [comment]
Done! :-)

**Pull request merged**